rocAL/rocAL/source/hw_jpeg_decoder.cpp

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+#ifdef RALI_VIDEO
+
+#include <stdio.h>
+#include <commons.h>
+#include "hw_jpeg_decoder.h"
+#include <boost/filesystem.hpp>
+
+namespace filesys = boost::filesystem;
+
+#define AVIO_CONTEXT_BUF_SIZE   32768     //32K
+
+struct buffer_data {
+    uint8_t *ptr;
+    size_t size; ///< size left in the buffer
+};
+
+static inline int num_hw_devices() {
+    char device[128] = "";
+    int num_hw_devices = 0;
+    for (int i=0; i<10000; i++){
```

Comment:
can you execute the  ls -l /dev/dri/by-path and then parse the output to find the number of renderD* devices?

Reply:
done

Reply:
@asalmanp, @kiritigowda : I have made the necessary changes. Can we merge this PR?

Reply:
@rrawther the changes look good, and we can merge this PR. Thank you.